app/org/sagebionetworks/bridge/play/controllers/SubpopulationController.java

```diff
@@ -0,0 +1,68 @@
+package org.sagebionetworks.bridge.play.controllers;
+
+import static org.sagebionetworks.bridge.Roles.DEVELOPER;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+
+import org.sagebionetworks.bridge.models.GuidVersionHolder;
+import org.sagebionetworks.bridge.models.accounts.UserSession;
+import org.sagebionetworks.bridge.models.studies.Study;
+import org.sagebionetworks.bridge.models.studies.Subpopulation;
+import org.sagebionetworks.bridge.services.SubpopulationService;
+
+import play.mvc.Result;
+
+@Controller
+public class SubpopulationController extends BaseController {
+
+    private SubpopulationService subpopService;
+    
+    @Autowired
+    public final void setSubpopulationService(SubpopulationService subpopService) {
+        this.subpopService = subpopService;
+    }
+
+    public Result getAllSubpopulations() {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        Study study = studyService.getStudy(session.getStudyIdentifier());
+        
+        List<Subpopulation> subpopulations = subpopService.getSubpopulations(study);
+        return okResult(subpopulations);
+    }
+    public Result createSubpopulation() throws Exception {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        Study study = studyService.getStudy(session.getStudyIdentifier());
+        
+        Subpopulation subpop = parseJson(request(), Subpopulation.class);
+        subpop = subpopService.createSubpopulation(study, subpop);
+        
+        return createdResult(new GuidVersionHolder(subpop.getGuid(), subpop.getVersion()));
+    }
+    public Result updateSubpopulation(String guid) {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+        Study study = studyService.getStudy(session.getStudyIdentifier());
+        
+        Subpopulation subpop = parseJson(request(), Subpopulation.class);
+        subpop.setGuid(guid);
```

Comment:
I feel like if they set the guid in the URL, which is to a resource, then that's what they're updating, and it's correct to ignore the guid in the JSON. It's not really even a documented property of the JSON object they are sending to us at that point, so this is just to prevent a security hole where they set it in a way we don't want. Like auto-marshaling a property to an object that's not supposed to be exposed.